foxglove_bridge_base/include/foxglove_bridge/websocket_server.hpp

```diff
@@ -0,0 +1,541 @@
+#pragma once
```

Comment:
🙁  can this file come from foxglove-websocket on conan instead of being vendored here? What were the changes needed?

Reply:
I guess we are going to run into the same packaging issues we did with mcap that led to mcap_vendor. Time for a foxglove_websocket_vendor?

Reply:
Not really, ROS (1 or 2) build systems don't include conan. The closest we could do is create a foxglove_websocket_vendor package similar to what we did with mcap_vendor, but it would need to be done for both ROS 1 and ROS 2 and would really be overkill compared to this approach.

Aside from that, this is not the same implementation. `foxglove-websocket` blocks when you call `run()`, while this is a multi-threaded server that asynchronously runs the WebSocket event loop when you call `start()`.

Reply:
What is our plan for dealing with bugfixes and protocol changes upstream in https://github.com/foxglove/ws-protocol/blob/main/cpp/foxglove-websocket/include/foxglove/websocket/server.hpp? Is that server implementation becoming abandonware? Are we maintaining two servers?